src/reply_parser.h

```diff
@@ -0,0 +1,61 @@
+
+#ifndef SRC_REPLY_PARSER_H_
+#define SRC_REPLY_PARSER_H_
+
+#include <stddef.h>
+
+typedef struct ReplyParser {
+    /* The current location on the reply buffer, need to set to the beginning of the reply*/
+    const char* curr_location;
+
+    /* Called when the parser reach an empty mbulk ('*-1') */
+    void (*null_array_callback)(void* ctx, const char* proto, size_t proto_len);
+
+    /* Called when the parser reach an empty bulk ('$-1') (bulk len is -1) */
+    void (*null_bulk_string_callback)(void* ctx, const char* proto, size_t proto_len);
+
+    /* Called when the parser reach a bulk ('$'), given the bulk payload and size */
+    void (*bulk_string_callback)(void* ctx, const char* str, size_t len, const char* proto, size_t proto_len);
+
+    /* Called when the parser reach an error ('-'), given the error message and len */
```

Comment:
```suggestion
    /* Called when the parser reaches an error ('-'), given the error message and len */
```

Repeats a few times below.